normandy/recipes/api/views.py

```diff
@@ -37,7 +37,21 @@ def update(self, request, *args, **kwargs):
 
         return super().update(request, *args, **kwargs)
 
-    @detail_route(permission_classes=[], renderer_classes=[JavaScriptRenderer])
-    def implementation(self, request, name=None):
-        obj = self.get_object()
-        return Response(obj.implementation)
+
+class ActionImplementationView(generics.RetrieveAPIView):
+    """
+    Retrieves the implementation code for an action. Raises a 404 if the
+    given hash doesn't match the has we've stored.
+    """
+    queryset = Action.objects.all()
+    lookup_field = 'name'
+
+    permission_classes = ()
+    renderer_classes = (JavaScriptRenderer,)
```

Comment:
Personally, I prefer to use lists here instead of tuples. I think Django 1.8 started recommending that for settings too. But it doesn't really matter.